src/charm.py

```diff
@@ -183,6 +299,21 @@ def _get_interfaces(self):
             raise CheckFailed(err, BlockedStatus)
         return interfaces
 
+    def main(self, event):
```

Comment:
Add doc string. This code is not touched, but we should be bringing it up to standard.